pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -42,9 +44,13 @@
 @ThreadSafe
 public abstract class BaseTableDataManager implements TableDataManager {
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseTableDataManager.class);
+  // cache atmost a given max of deleted segments
+  private static final int MAX_DELETED_SEGMENT_CACHE_SIZE = 1000;
```

Comment:
Discussed offline. We can probably add the time based eviction and increase `MAX_DELETED_SEGMENT_CACHE_SIZE`.

Reply:
Updated.